vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1863,6 +1866,15 @@ def initialize_kv_cache(self, kv_cache_config: KVCacheConfig) -> None:
                     # KV cache specs.
                     raise ValueError("Unknown KV cache spec type.")
 
+        if self.speculative_config and self.speculative_config.use_eagle():
+            assert isinstance(self.drafter, EagleProposer)
+            assert len(
+                set([
+                    kv_cache_group_ids[layer_name]
+                    for layer_name in self.drafter.attn_layer_names
+                ])) == 1, "For multi-layer eagle draft model, "
+            "all layers should belong to the same kv cache group"
+
```

Comment:
Can we put these assertions for eagle to a function inside eagle proposer?

Reply:
can we add a function like `self.drafter.verify_same_kv_cache_group(kv_cache_config)`?

Reply:
@heheda12345 Done!